internal/controller/state/dataplane/configuration_test.go

```diff
@@ -4895,3 +4895,62 @@ func TestBuildNginxPlus(t *testing.T) {
 		})
 	}
 }
+
+func TestBuildWorkerConnections(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		gw                   *graph.Gateway
+		msg                  string
+		expWorkerConnections int32
+	}{
+		{
+			msg:                  "NginxProxy is nil",
+			gw:                   &graph.Gateway{},
+			expWorkerConnections: 1024,
```

Comment:
Let's use the default value here instead of a literal value. That way if the default changes, we don't need to update the tests.